third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -173,6 +174,44 @@ impl InterpreterImpl {
         }
     }
 
+    fn load_generic_function(
+        &mut self,
+        resolver: &Resolver,
+        current_frame: &Frame,
+        gas_meter: &mut impl GasMeter,
+        idx: FunctionInstantiationIndex,
+    ) -> VMResult<LoadedFunction> {
+        let ty_args = resolver
+            .instantiate_generic_function(Some(gas_meter), idx, current_frame.function.ty_args())
```

Comment:
I somehow thought you would use the subframes to avoid this if available. To solve the `while (c) f<T>(x)` perf problem.

Reply:
Ah ok, you only call this function the first time.

Might be perhaps not to bad to encapsulate the entire caching logic here instead of the execute function.